crates/ide/src/runnables.rs

```diff
@@ -183,8 +183,9 @@ pub(crate) fn runnables(db: &RootDatabase, file_id: FileId) -> Vec<Runnable> {
     });
 
     sema.file_to_module_defs(file_id)
-        .map(|it| runnable_mod_outline_definition(&sema, it))
-        .for_each(|it| add_opt(it, None));
+        .filter_map(|it| runnable_mod_outline_definition(&sema, it))
+        .filter(|it| it.nav.file_id == file_id)
```

Comment:
I think we need to just discard the assert actually? The problem here is the upmapping of `include` yielding back a real file id isntead of the usual macro call, and so I think we can still hit this with some silly stuff like
```rs
impl Type {
    include!("path/of/a-runnable-assoc-thing)
}
```

The assert used to be right when it was written but since we've changed how token upmapping works for includes so thats no longer the case I think (we had a similar issue with diagnostics)

Reply:
Agree. I also find the assertion here a bit strange.

Reply:
The reason for it was mainly if we ended up generating a runnable for a differing file that indicates a bug, since we are supposed to generate them for a specific file, but with `include` this can happen either way so plain filtering is the way to go now